### PR TITLE
ENH: `max_rows` `max_cols` kw for `Styler.to_html` + `styler.options`

### PR DESCRIPTION
**Pull request by attack68**
one more item towards deprecating `DataFrame.to_html` for #41693 

- [x] adds options to config
- [x] update `options.rst`
- [x] add tests
- [x] update to_html docstring
- [x] whats new
- [ ] update user guide (will be follow on after all options in master tracker added)   

and an item towards consistent `pandas.options` for #41395

Screen Shot 2021-08-29 at 09 26 45

### CODE REVIEW

File: pandas/io/formats/style_render.py [hunk 1]
```diff
@@ -836,6 +871,11 @@ def scale_down(rn, cn):
         else:
             return int(rn * scaling_factor), cn
 
+    if max_rows:
```
Comment (jreback) [inline pandas/io/formats/style_render.py:874]: this exercised enough in tests?

**Review by jreback — CHANGES_REQUESTED** [review]
can you rebase

**Comment by jreback** [comment]
can you merge master

File: pandas/io/formats/style_render.py (hunk 1 above)
Comment (attack68) [reply pandas/io/formats/style_render.py:892]: added specific cases

**Review by jreback — APPROVED** [review]

**Pull request merged**